### PR TITLE
refactor(core): break manager struct into smaller ones

### PR DESCRIPTION
**Pull request by lucasfernog**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: core/tauri/src/manager/window.rs [hunk 1]
```diff
@@ -0,0 +1,810 @@
...
+/// Uses a custom URI scheme handler to resolve file requests
+pub struct UriSchemeProtocol<R: Runtime> {
+  /// Handler for protocol
+  #[allow(clippy::type_complexity)]
+  pub protocol:
+    Box<dyn Fn(&AppHandle<R>, http::Request<Vec<u8>>, UriSchemeResponder) + Send + Sync>,
+}
+
+#[cfg(desktop)]
+pub struct MenuManager<R: Runtime> {
```
Comment (amrbashir) [inline core/tauri/src/manager/window.rs]: I think `MenuManager` could also be split into its own file and should be a field in `AppManager` instead as it also affects `App/AppHandle`.

File: core/tauri/src/manager/mod.rs [hunk 2]
```diff
@@ -0,0 +1,792 @@
...
+  /// The asset bytes.
+  pub bytes: Vec<u8>,
+  /// The asset's mime type.
+  pub mime_type: String,
+  /// The `Content-Security-Policy` header value.
+  pub csp_header: Option<String>,
+}
+
+#[default_runtime(crate::Wry, wry)]
+pub struct AppManager<R: Runtime> {
```
Comment (amrbashir) [inline core/tauri/src/manager/mod.rs:180]: should probably also move this to `manager/app.rs`

**Review by amrbashir — CHANGES_REQUESTED** [review]
I think my main problem with this PR is the excessive usage of `Arc`, do you think we can somehow remove the need for it?

Edit: the `Arc` usage was excessive before as well, I guess I misspoke, what I mean, can we reduce the `Arc` usage now that we have grouped multiple fields together?

File: core/tauri/src/manager/window.rs (hunk 1 above)
Comment (lucasfernog) [reply core/tauri/src/manager/window.rs]: good one, pushed

File: core/tauri/src/manager/mod.rs (hunk 2 above)
Comment (lucasfernog) [reply core/tauri/src/manager/mod.rs:180]: i'm not really sure, since AppManager is the root manager it can live in mod.rs IMO

**Comment by lucasfernog** [comment]
@amrbashir I've done my best with the Arc removals, some are still around because there's some types that are cloned and moved to another thread / closure (mostly callbacks).

File: core/tauri/src/manager/mod.rs (hunk 2 above)
Comment (amrbashir) [reply core/tauri/src/manager/mod.rs:180]: Yeah I agree but app.rs gives more clarity as it is not clear from first glance where AppManager lives.

Comment (amrbashir) [reply core/tauri/src/manager/mod.rs:180]: anyways, this is just a nit so go with your gut.

**Review by amrbashir — APPROVED** [review]

**Pull request merged**